assets/js/base/components/text-input/text-input.tsx

```diff
@@ -87,6 +88,7 @@ const TextInput = forwardRef< HTMLInputElement, TextInputProps >(
 							: ariaDescribedBy
 					}
 					required={ required }
+					{ ...rest }
```

Comment:
What was the goal of doing `{ ...rest }`? this is quite the change and exposes a lot of implementation detail to the user.
It also means the user might overwrite several events like onChange, onFocus, and onBlur.
You should make sure to merge events or just be wary of what you're exposing.

Reply:
It was so we could have step, min, and max values on the input field, an alternative we could offer is to just check if those specific props are passed, check if the type is number then spread only those three onto the input 🤔 let me try that.

Reply:
Updated with this change. Seems to be working fine